articles/quickstart/webapp/nodejs/01-login.md

```diff
@@ -14,4 +16,239 @@ github:
+passport.serializeUser(function(user, done) {
+  done(null, user);
+});
+
+passport.deserializeUser(function(user, done) {
+  done(null, user);
+});
+```
+
+## Implementing Login, User Profile and Logout
+
+### Adding the Routes
+
+In this example, we will implement the following routes:
+
+* a route `/login`, that will trigger authentication by calling Passport's `authenticate` method. The user will be redirected to the login page.
+* a route `/callback`, that will be finalize the authentication when the user returns from the Auth0 login page and redirects to the profile page on sucessful login.
+* a `/user` route, to display the user's profile.
+* a route `/logout`, that closes the local user session and redirects the user again to the root index `/`.
+
+In the authentication step, make sure to pass the scope parameter with values `openid email profile` in order to access email and the other attributes stored in the user profile within the tenant. This will be needed in order to display the user's information on the profile page.
+
+The user's profile route should only be accessible if the user is logged in. Use the `ensureLoggedIn` middleware for this purpose. To have full access to the user profile on `userProfile`, stringify the `user` object.
+
+:::note
+This tutorial implements logout by closing the local user session. After logging out, the user will continue to have an open session at the tenant level. For other implementations please refer to [our logout docs](https://auth0.com/docs/logout). 
+:::
+
+Create a new router `routes/auth.js`:
+
+```js
+// routes/auth.js
+
+var express = require('express');
+var router = express.Router();
+var Auth0Strategy = require('passport-auth0'),
+    passport = require('passport');
+var ensureLoggedIn = require('connect-ensure-login').ensureLoggedIn();
+
+// Perform the login, after login Auth0 will redirect to callback
+router.get('/login',
+  passport.authenticate('auth0', {scope: 'openid email profile'}), function (req, res) {
+  res.redirect("/");
+});
+
+// Perform the final stage of authentication and redirect to '/user'
+router.get('/callback',
+  passport.authenticate('auth0', { failureRedirect: '/login' }),
+  function(req, res) {
+    if (!req.user) {
+      throw new Error('user null');
+    }
+    res.redirect("/user");
+  }
+);
+
+/* GET user profile. */
+router.get('/user', ensureLoggedIn, function(req, res, next) {
+  res.render('user', {
+    user: req.user ,
+    userProfile: JSON.stringify(req.user, null, '  ')
+  });
+});
+
+// Perform session logout and redirect to homepage
+router.get('/logout', (req, res) => {
+  req.logout();
+  res.redirect('/');
+});
+
+module.exports = router;
+```
+
+Include this new router in your app, by requiring the file in the top section of  `app.js`:
+
+```js
+// app.js
+
+var indexRouter = require('./routes/index');
+var authRouter = require('./routes/auth');
+
+//..
+//..
+
+app.use('/', indexRouter);
+//..
+app.use('/', authRouter);
+//..
+```
+
```

Comment:
Maybe we need to explain the image (e.g. run the app, click 'Login' and you'll see this)